pkg/route/graph/analysis/analysis.go

```diff
@@ -63,3 +67,23 @@ route:
 
 	return markers
 }
+
+func FindPathBasedPassthroughRoutes(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, uncastRouteNode := range g.NodesByKind(routegraph.RouteNodeKind) {
+		routeNode := uncastRouteNode.(*routegraph.RouteNode)
+
+		if len(routeNode.Spec.Path) > 0 && routeNode.Spec.TLS != nil && routeNode.Spec.TLS.Termination == routeapi.TLSTerminationPassthrough {
+			markers = append(markers, osgraph.Marker{
+				Node: routeNode,
+
+				Severity: osgraph.WarningSeverity,
+				Key:      PathBasedPassthroughWarning,
+				Message:  fmt.Sprintf("%s is path-based and passthrough terminated which is an invalid combo.", routeNode.ResourceString()),
```

Comment:
Maybe "%s is path-based and uses passthrough termination, which is an invalid combination.

definitely don't use the word combo, its too casual.